agent/acs/handler/attach_eni_handler_test.go

```diff
@@ -253,21 +255,75 @@ func TestENIAckSingleMessage(t *testing.T) {
 	assert.Equal(t, aws.StringValue(eniAckRequested.MessageId), eniMessageId)
 }
 
-// TestENIAckForMessageIdMismatch tests for mismatched messageId's in the
+// TestENIAckSingleMessageDuplicateENIAttachmentMessageStartsTimer checks the ack for a single message
+// and ensures that the ENI ack expiration timer is started
+func TestENIAckSingleMessageDuplicateENIAttachmentMessageStartsTimer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockState := mock_dockerstate.NewMockTaskEngineState(ctrl)
+	manager := mock_statemanager.NewMockStateManager(ctrl)
+
+	ctx := context.TODO()
+	mockWSClient := mock_wsclient.NewMockClientServer(ctrl)
+	eniAttachHandler := newAttachENIHandler(ctx, clusterName, containerInstanceArn, mockWSClient, mockState, manager)
+
+	var eniAckRequested *ecsacs.AckRequest
+	gomock.InOrder(
+		mockWSClient.EXPECT().MakeRequest(gomock.Any()).Do(func(ackRequest *ecsacs.AckRequest) {
+			eniAckRequested = ackRequest
+			eniAttachHandler.stop()
+		}),
+		// Sending an empty attachment results in an error in starting the time.
+		// Ensuring that statemanager.Save() is not invoked should be a strong
+		// enough check to ensure that the timer was started
+		mockState.EXPECT().ENIByMac(randomMAC).Return(&api.ENIAttachment{}, true),
+		manager.EXPECT().Save().Return(nil).Times(0),
+	)
+	go eniAttachHandler.start()
+
+	mockNetInterface1 := ecsacs.ElasticNetworkInterface{
+		Ec2Id:         aws.String("1"),
+		MacAddress:    aws.String(randomMAC),
+		AttachmentArn: aws.String("attachmentarn"),
+	}
+	message := &ecsacs.AttachTaskNetworkInterfacesMessage{
+		MessageId:            aws.String(eniMessageId),
+		ClusterArn:           aws.String(clusterName),
+		ContainerInstanceArn: aws.String(containerInstanceArn),
+		ElasticNetworkInterfaces: []*ecsacs.ElasticNetworkInterface{
+			&mockNetInterface1,
+		},
+		TaskArn:       aws.String(taskArn),
+		WaitTimeoutMs: aws.Int64(waitTimeoutMillis),
+	}
+
+	// Expect an error starting the timer because of <=0 duration
```

Comment:
To make this comments easier to understand, can you add a negative value of WaitTimeoutMs in the returned ENIAttachment in line 280?